src/transformers/generation/watermarking.py

```diff
@@ -181,43 +189,40 @@ def __call__(
         self,
         input_ids: torch.LongTensor,
         z_threshold: float = 3.0,
-        return_prediction: bool = True,
         return_dict: bool = False,
     ) -> Union[WatermarkDetectorOutput, np.array]:
         """
         Args:
         input_ids (`torch.LongTensor`):
             The watermark generated text. It is advised to remove the prompt, which can affect the detection.
-        z_threshold (`Dict`, *optional*):
-            Changing this threshold will change the sensitivity of the detector.
-        return_prediction (`bool`):
-            Whether to return user-friendly predictions as a boolean array or not.
-        return_dict (`bool`):
-            Whether to return `~generation.WatermarkDetectorOutput` or not. If not it will return `z_score` if
-            `return_prediction=False` else `prediction`,
+        z_threshold (`Dict`, *optional*, defaults to `3.0`):
+            Changing this threshold will change the sensitivity of the detector. Higher z threshold gives less
+            sensitivity and vice versa for lower z threshold.
+        return_dict (`bool`,  *optional*, defaults to `False`):
+            Whether to return `~generation.WatermarkDetectorOutput` or not. If not it will return boolean predictions,
 
         Return:
-            [`~generation.WatermarkDetectorOutput`] or `np.array`: A [`~utils.ModelOutput`] (if `return_dict=True`
-            otherwise a `np.array`.
+            [`~generation.WatermarkDetectorOutput`] or `np.array`: A [`~generation.WatermarkDetectorOutput`]
+            if `return_dict=True` otherwise a `np.array`.
 
         """
 
+        # Let's assume that if one batch start with `bos`, all batched also do
```

Comment:
Question: what happens with left-padding? Does left-padding have an impact on the detector?

Reply:
Hmm, I did not think someone would feed left-padded text. What I usually do if feeding only the generated text part, excluding the prompt.
Prompt itself has some effect on the z-score but in my toy examples the final prediction did not change. But I believe long prompt with smaller generated text might be predicted as "human" 🤔 

Anyway, I will better explain this in the docs, adding more information next to the "Generation strategies".